File: packages/live-update/android/src/main/java/io/capawesome/capacitorjs/plugins/liveupdate/classes/results/GetVersionCodeResult.java
package io.capawesome.capacitorjs.plugins.liveupdate.classes.results;

import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.liveupdate.interfaces.Result;
import org.json.JSONObject;

public class GetVersionCodeResult implements Result {

    private String versionCode;

    public GetVersionCodeResult(String versionCode) {
        this.versionCode = versionCode;
    }

    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("versionCode", versionCode);
        return result;
    }
}

File: packages/live-update/ios/Plugin/Classes/Results/GetVersionCodeResult.swift
import Foundation
import Capacitor

@objc public class GetVersionCodeResult: NSObject, Result {
    let versionCode: String

    init(versionCode: String) {
        self.versionCode = versionCode
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["versionCode"] = versionCode
        return result as AnyObject
    }
}
